test/csuite/CMakeLists.txt

```diff
@@ -54,36 +56,40 @@ define_c_test(
         random_directio/main.c
         random_directio/util.c
     DIR_NAME random_directio
-    SMOKE
+    EXEC_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/random_directio/smoke.sh
+    ARGUMENTS $<TARGET_FILE:test_random_directio>
     DEPENDS "WT_POSIX"
 )
 
 define_c_test(
     TARGET test_rwlock
     SOURCES rwlock/main.c
     DIR_NAME rwlock
+    ARGUMENTS -h $<SHELL_PATH:$<TARGET_FILE_DIR:test_rwlock>/WT_HOME>
```

Comment:
Should we have `EXEC_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/rwlock/smoke.sh` here ?

Reply:
Good question! We don't need to define an `EXEC_SCRIPT` for the `rwlock` test, since it isn't executed with a `smoke.sh` script i.e. the test doesn't have a `smoke.sh` script (both in CMake and autconf). Rather the test binary is just executed directly. 
If we don't pass an `EXEC_SCRIPT` to the `define_c_test`, this essentially means the test will be run by just executing the binary.    
(As most of the `smoke.sh` scripts are designed to just call the binary multiple times with different arguments, I'd eventually like to covert the majority of the c tests definitions to not depend on a wrapper script, since its not super portable on Windows. The `VARIANTS` feature I also added in this PR provides the framework to assist this)